cmd/ocm/list/region/cmd.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+  http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package region
+
+import (
+	"fmt"
+	"os"
+	"text/tabwriter"
+
+	"github.com/openshift-online/ocm-cli/pkg/arguments"
+	"github.com/openshift-online/ocm-cli/pkg/cluster"
+	"github.com/openshift-online/ocm-cli/pkg/ocm"
+	"github.com/openshift-online/ocm-cli/pkg/provider"
+	"github.com/spf13/cobra"
+)
+
+var args struct {
+	provider string
+	ccs      cluster.CCS
+}
+
+var Cmd = &cobra.Command{
+	Use:     "regions --provider=CLOUD_PROVIDER [--ccs CCS FLAGS]",
+	Aliases: []string{"regions"},
+	Short:   "List known/available cloud provider regions",
+	Long: "List regions of a cloud provider.\n\n" +
+		"In --ccs mode, fetch regions that would be available to *your* cloud account\n" +
```

Comment:
Requiring the `--ccs` flag here feels a bit strange. I think that if the user sends in the `--aws-access-key-id` and `--aws-secret-access-key` then we should just query for the regions with those credentials, regardless of `ccs.Enabled`. What do you think?

Reply:
In this command that makes some sense, but we do require --ccs in `ocm create cluster`.
I'm not sure I want to change creation to enable CCS implicitly, as CCS is a major decision to make?  WDYT?

But it's not good that we now silently ignore `--aws-access-key-id` when --ccs is not provided (in both create and list regions).
Let me fix that to give an explicit error suggesting `--ccs`...

Reply:
Now prints descriptive errors:
```
$ go run ./cmd/ocm create cluster cben-fail --aws-account-id=foo --aws-secret-access-key=mellon
Error: --aws-account-id, --aws-secret-access-key flags are meaningless without --ccs
exit status 1

$ go run ./cmd/ocm list regions --provider=aws --aws-secret-access-key=mellon
Error: --aws-secret-access-key flag is meaningless without --ccs
exit status 1
```
Is that better?

Reply:
@vkareh @igoihman PTAL.  Need a decision here, options:

- [ ] Make --ccs implicit even for `create cluster` when provider account flags given?
- [ ] Require explicit --ccs in `create cluster`, implicit in `list regions`?
- [x] Require explicit --ccs flag in both?

Reply:
explicit --ccs flag